op-service/txplan/txplan.go

```diff
@@ -285,6 +294,39 @@ func WithAgainstLatestBlock(cl AgainstLatestBlock) Option {
 	}
 }
 
+// Reader uses eth_call to view(read) the blockchain, and does not write in perspective of chain.
+// The chain will return byte strings when viewing, and does not return receipt.
+type Reader interface {
+	Call(ctx context.Context, msg ethereum.CallMsg) ([]byte, error)
+}
+
+func WithReader(cl Reader) Option {
+	return func(tx *PlannedTx) {
+		tx.Read.DependOn(
+			&tx.Sender,
+			&tx.To,
+			&tx.GasFeeCap,
+			&tx.GasTipCap,
+			&tx.Value,
+			&tx.Data,
+		)
```

Comment:
```suggestion
			&tx.Data,
			&tx.AccessList,
		)
```

I think you forgot the access-list dependency here